Assets/Scripts/Framework/UI/TextUpdaters/TextUpdater.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+namespace DerailedDeliveries.Framework.UI.TextUpdaters
+{
+    /// <summary>
+    /// A class responsible for updating text with tags.
+    /// </summary>
+    [RequireComponent(typeof(TextMeshProUGUI))]
+    public class TextUpdater : MonoBehaviour
+    {
+        [SerializeField]
+        private string _tag;
+
+        private TextMeshProUGUI _text;
+        private string _defaultText;
+        private Dictionary<string, string> _tags = new();
+
+        private protected virtual void Awake()
```

Comment:
Why is this awake method delclared as `protected virtual`? It can just be private right?

Reply:
No, if it is just private we can't override it from child classes.

Reply:
Sure, it wasn't really clear that classes could inherit TexUpdater.